### PR TITLE
Add raise to vyper, remove throw

### PR DESCRIPTION
**Pull request by charles-cooper**
### What I did
Implement #1342. Opening PR for feedback on semantics.

#### Todo:
- [x] docs
- [x] tests

### How I did it
Add ast case for raise, factor out `_assert_reason`

### How to verify it
```python
## raise1.vy
@public
def foo(bar: uint256):
  raise "Hello"

## raise2.vy
@public
def foo(bar: uint256):
  raise 1 # structure exception
```

### Description for the changelog
Remove throw from vyper
Add raise to vyper.

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Comment by charles-cooper** [comment]
Not sure why this doesn't pass lint on CI, it passes on my local machine. Does the makefile rule need to be updated to match CI?

File: vyper/parser/stmt.py [hunk 1]
```diff
@@ -441,27 +446,21 @@ def call(self):
...
-            method_id = fourbytes_to_int(sha3(b"Error(string)")[:4])
-            assert_reason = [
+    def _assert_reason(self, test_expr, msg):
+        if not isinstance(msg, ast.Str):
+            raise StructureException(
+                    'Reason parameter of assert needs to be a literal string.',
+                    msg)
+        if len(msg.s.strip()) == 0:
+            raise StructureException('Empty reason string not allowed.',
+                    self.stmt)
```
Comment (fubuloubu) [inline vyper/parser/stmt.py]: `vyper/parser/stmt.py:456:21: E128 continuation line under-indented for visual indent` is the linter exception. Does this even need to be on a new line?

**Comment by jacqueswww** [comment]
@charles-cooper weird, yes maybe we need to change `make lint` to `tox -e lint` instead?

File: vyper/parser/stmt.py (hunk 1 above)
Comment (charles-cooper) [reply vyper/parser/stmt.py]: 80 chars

**Comment by charles-cooper** [comment]
> @charles-cooper weird, yes maybe we need to change `make lint` to `tox -e lint` instead?

that did the trick

**Comment by charles-cooper** [comment]
opening a separate PR for that

**Comment by charles-cooper** [comment]
This PR is ready (removed the WIP tag)

**Pull request merged**